core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -468,7 +471,8 @@ public PropServerConfiguration(
         this.walRecreateDistressedSequencerAttempts = getInt(properties, env, PropertyKey.CAIRO_WAL_RECREATE_DISTRESSED_SEQUENCER_ATTEMPTS, 3);
         this.walSupported = getBoolean(properties, env, PropertyKey.CAIRO_WAL_SUPPORTED, true);
         this.walSegmentRolloverRowCount = getLong(properties, env, PropertyKey.CAIRO_WAL_SEGMENT_ROLLOVER_ROW_COUNT, 200_000);
-        this.walCommitSquashRowLimit = getInt(properties, env, PropertyKey.CAIRO_WAL_COMMIT_SQUASH_ROW_LIMIT, 512 * 1024);
+        this.walCommitSquashRowLimit = getInt(properties, env, PropertyKey.CAIRO_WAL_COMMIT_SQUASH_ROW_LIMIT, 100 * 1024 * 1024);
```

Comment:
This change is not what's documented in `server.conf` plus using 100M seems to be too high value. For a single `long` column this means over 800MB of RAM, so OOMs and friends.

Reply:
good catch, @ideoma what do you think ?

Reply:
WAL lag is not in RAM most of the time, it's in memory-mapped files. It's brought into memory before copying to into the partitions only.

This is the last measure to hold the data in lag when TableWriter knows that the commit is unavoidable O3 and this is to reduce the number of O3 commits. If O3 is not detected all data is committed to the table, and nothing is kept in the lag.

I can reduce it, perhaps 10M will be good enough

Reply:
Decreasing it to a low value will not obviously save memory.
Yes, O3 data will be smaller, but the memory we use for O3 commit is usually O3 data + partition size. The bigger is the partition the more memory is used to merge.

Setting this option to 1M can result in 10x O3 merges and the last merge will lift 10M rows. Same memory as if the parameter is set to 10M and then instead of 10x O3 it will be a single O3 commit of 10M rows.

Reply:
Does the default value of 100M mean that we expect partitions to be 100M? I get the point on partition size, but in many cases O3 data would contain rows from multiple partitions.

Reply:
> I changed the default to 10M.

Could you also update the default value in `core/src/main/resources/io/questdb/site/conf/server.conf`?